dask/array/tests/test_reductions.py

```diff
@@ -158,9 +162,11 @@ def reduction_2d_test(da_func, darr, np_func, narr, use_dtype=True, split_every=
     assert same_keys(da_func(darr, axis=1), da_func(darr, axis=1))
     assert same_keys(da_func(darr, axis=(1, 0)), da_func(darr, axis=(1, 0)))
 
-    if use_dtype:
-        assert_eq(da_func(darr, dtype="f8"), np_func(narr, dtype="f8"))
-        assert_eq(da_func(darr, dtype="i8"), np_func(narr, dtype="i8"))
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore", np.ComplexWarning)
+        if use_dtype:
```

Comment:
Would it make sense to swap these blocks out?

```suggestion
    if use_dtype:
        with warnings.catch_warnings():
            warnings.simplefilter("ignore", np.ComplexWarning)
```